contracts/staking/IStaking.sol

```diff
@@ -54,25 +57,29 @@ interface IStaking is
 
     function createStake(NftId target, uint256 dipAmount) external returns(NftId id);
     function stake(NftId id, uint256 dipAmount) external;
+    function createStakeWithSignature(address owner, NftId target, uint256 dipAmount, bytes32 signatureId, bytes calldata signature) external returns(NftId stakeId);
+    function restake(NftId id, NftId newTarget) external;
+    function restakeWithSignature(address owner, NftId stakeId, NftId newTarget, bytes32 signatureId, bytes calldata signature) external;
```

Comment:
```suggestion
    function createStakeWithSignature(address owner, NftId targetNftId, uint256 dipAmount, bytes32 signatureId, bytes calldata signature) external returns(NftId stakeId);
    function restake(NftId nftId, NftId newTargetNftId) external;
    function restakeWithSignature(address owner, NftId stakeNftId, NftId newTargetNftId, bytes32 signatureId, bytes calldata signature) external;
```

We should make the attribute names a bit more expressive so its clearer what they are (e.g. when you don't have the custom types in typescript any more)